clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -287,6 +287,7 @@ private KafkaProducer(ProducerConfig config, Serializer<K> keySerializer, Serial
             this.sender = new Sender(client,
                     this.metadata,
                     this.accumulator,
+                    config.getInt(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION) == 1,
```

Comment:
That sounds good to me as well - I don't recollect an earlier diff with that approach. @becketqin ?